[bug]:  Asterisk 18 GIT fails to build with app_voicemail and app_voicemail_odbc error

### Severity

Major

### Versions

18 GIT

### Components/Modules

app_voicemail app_voicemail_odbc

### Operating Environment

CentOS 7

### Frequency of Occurrence

None

### Issue Description

Unable to compile Asterisk 18 GIT due a coding inssue in app_voicemail and app_voicemail_odbc



### Relevant log output

```shell
[CCi] app_voicemail.i -> app_voicemail.o
app_voicemail.c: In function ‘copy_message’:
app_voicemail.c:6395:3: warning: implicit declaration of function ‘SCOPE_CALL_WITH_INT_RESULT’ [-Wimplicit-function-declaration]
   exists = SCOPE_CALL_WITH_INT_RESULT(-1, EXISTS, fromdir, msgnum, frompath, chan ? ast_channel_language(chan) : "");
   ^
app_voicemail.c:6395:43: error: ‘EXISTS’ undeclared (first use in this function)
   exists = SCOPE_CALL_WITH_INT_RESULT(-1, EXISTS, fromdir, msgnum, frompath, chan ? ast_channel_language(chan) : "");
                                           ^
app_voicemail.c:6395:43: note: each undeclared identifier is reported only once for each function it appears in
app_voicemail.c: In function ‘close_mailbox’:
app_voicemail.c:9665:41: error: ‘EXISTS’ undeclared (first use in this function)
    res = SCOPE_CALL_WITH_INT_RESULT(-1, EXISTS, vms->curdir, x, vms->fn, NULL);
                                         ^
app_voicemail.c: At top level:
cc1: warning: unrecognized command line option "-Wno-format-truncation" [enabled by default]
make[1]: *** [app_voicemail.o] Error 1
make: *** [apps] Error 2

-------------------- app_voicemail_odbc ------------------------------

app_voicemail_odbc.c: In function ‘copy_message’:
app_voicemail_odbc.c:6395:3: warning: implicit declaration of function ‘SCOPE_CALL_WITH_INT_RESULT’ [-Wimplicit-function-declaration]
   exists = SCOPE_CALL_WITH_INT_RESULT(-1, EXISTS, fromdir, msgnum, frompath, chan ? ast_channel_language(chan) : "");
   ^
app_voicemail_odbc.c:6395:43: error: ‘EXISTS’ undeclared (first use in this function)
   exists = SCOPE_CALL_WITH_INT_RESULT(-1, EXISTS, fromdir, msgnum, frompath, chan ? ast_channel_language(chan) : "");
                                           ^
app_voicemail_odbc.c:6395:43: note: each undeclared identifier is reported only once for each function it appears in
app_voicemail_odbc.c: In function ‘leave_voicemail’:
app_voicemail_odbc.c:7185:48: error: ‘RETRIEVE’ undeclared (first use in this function)
   int success = SCOPE_CALL_WITH_INT_RESULT(-1, RETRIEVE, prefile, -1, ext, context);
                                                ^
app_voicemail_odbc.c: In function ‘close_mailbox’:
app_voicemail_odbc.c:9665:41: error: ‘EXISTS’ undeclared (first use in this function)
    res = SCOPE_CALL_WITH_INT_RESULT(-1, EXISTS, vms->curdir, x, vms->fn, NULL);
                                         ^
app_voicemail_odbc.c: At top level:
cc1: warning: unrecognized command line option "-Wno-format-truncation" [enabled by default]
make[1]: *** [app_voicemail_odbc.o] Error 1
make: *** [apps] Error 2
   [CPP] app_directory.c -> app_directory.i
   [CCi] app_directory.i -> app_directory.o
   [LD] app_directory.o -> app_directory.so
   [CCi] app_voicemail_odbc.i -> app_voicemail_odbc.o
app_voicemail_odbc.c: In function ‘copy_message’:
app_voicemail_odbc.c:6395:3: warning: implicit declaration of function ‘SCOPE_CALL_WITH_INT_RESULT’ [-Wimplicit-function-declaration]
   exists = SCOPE_CALL_WITH_INT_RESULT(-1, EXISTS, fromdir, msgnum, frompath, chan ? ast_channel_language(chan) : "");
   ^
app_voicemail_odbc.c:6395:43: error: ‘EXISTS’ undeclared (first use in this function)
   exists = SCOPE_CALL_WITH_INT_RESULT(-1, EXISTS, fromdir, msgnum, frompath, chan ? ast_channel_language(chan) : "");
                                           ^
app_voicemail_odbc.c:6395:43: note: each undeclared identifier is reported only once for each function it appears in
app_voicemail_odbc.c: In function ‘leave_voicemail’:
app_voicemail_odbc.c:7185:48: error: ‘RETRIEVE’ undeclared (first use in this function)
   int success = SCOPE_CALL_WITH_INT_RESULT(-1, RETRIEVE, prefile, -1, ext, context);
                                                ^
app_voicemail_odbc.c: In function ‘close_mailbox’:
app_voicemail_odbc.c:9665:41: error: ‘EXISTS’ undeclared (first use in this function)
    res = SCOPE_CALL_WITH_INT_RESULT(-1, EXISTS, vms->curdir, x, vms->fn, NULL);
                                         ^
app_voicemail_odbc.c: At top level:
cc1: warning: unrecognized command line option "-Wno-format-truncation" [enabled by default]
make[1]: *** [app_voicemail_odbc.o] Error 1
make: *** [apps] Error 2
```


### Asterisk Issue Guidelines

- [X] Yes, I have read the Asterisk Issue Guidelines

--- a/include/asterisk/logger.h
+++ b/include/asterisk/logger.h
@@ -1020,6 +1020,9 @@ unsigned long _ast_trace_dec_indent(void);
 #define SCOPE_CALL_WITH_RESULT(level, __var, __funcname, ...) \
 	__var = __funcname(__VA_ARGS__)
 
+#define SCOPE_CALL_WITH_INT_RESULT(level, __funcname, ...) \
+	__funcname(__VA_ARGS__)
+
 #define SCOPE_EXIT(...) \
 	ast_debug(__scope_level, " " __VA_ARGS__)
 

